utils/src/main/scala/za/co/absa/enceladus/utils/validation/field/BinaryFieldValidator.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.validation.field
+
+import java.util.Base64
+
+import za.co.absa.enceladus.utils.implicits.StructFieldImplicits._
+import za.co.absa.enceladus.utils.schema.{MetadataKeys, MetadataValues}
+import za.co.absa.enceladus.utils.types.TypedStructField
+import za.co.absa.enceladus.utils.validation.{ValidationError, ValidationIssue}
+
+import scala.util.{Failure, Success, Try}
+
+object BinaryFieldValidator extends BinaryFieldValidator
+
+class BinaryFieldValidator extends FieldValidator {
```

Comment:
Please refactor this class.
It's strangely inconsistently written. Functions for simple one liners used ones, `vals` for trivial expressions, on the other hand complicated expressions are directly operated on...

Reply:
Redone, it's just an object now. I have seen it with `NumericFieldValidator` and the likes, so I just followed the suit.